### ISSUE TITLE
onAuthStateChange should fire on initial load

### ISSUE THREAD

**Issue by mikob** [supabase/supabase#3513]
# Feature request

## Is your feature request related to a problem? Please describe.
Currently it seems there is no deterministic way to know if a supabase session is done loading.

If the user is signed out,`auth.session()` will return `null` and onAuthStateChange won't fire. If the user is signed-in, `auth.session()` will return `null`, but then `onAuthStateChange` will fire soon after with the session data.

Consider this code:
```
        const sess = supabase.auth.session();
        log("session event", sess);

        changeSession(sess);

        supabase.auth.onAuthStateChange((_event, session) => {
            log("session event", _event);
            changeSession(session);
        });
```

A clear and concise description of what you want and what your use case is.
I need to load certain user history with knowledge of whether the user is signed-in or not. Yes it's possible to first fire assuming they aren't logged in, and then backtracking and firing again (once onAuthStateChange is hit) saying actually they are signed in, but it forces me to fire changeSession a function which is resource intensive on my end twice. 

## Describe the solution you'd like

Either onAuthStateChange should fire initially by default (how firebase auth works BTW), whether the user is signed in or not (perhaps a special INITIAL event) or should take a parameter {initial: true} to allow this behavior.

**Comment by lsrugo** [comment]
I would love to see something like this. I want to redirect the user to a login page if they are not signed in but I am having a hard time checking if they are actually signed in or not

**Comment by kiwicopple** [comment]
Yes agreed. I'll tag this one with a "help wanted" - if anyone is interested, this can be implemented in gotrue-js after the [recoverSession()](https://github.com/supabase/gotrue-js/blob/4d57cb5db2c833bb08dcb72e1ce025615624efdd/src/GoTrueClient.ts#L89) fucntion call

**Comment by megacherry** [comment]
are there any news regarding the pull request? i didn't want to ask in the pull request.

**Comment by Luc069** [comment]
This is the main problem in the supabase world
